### ISSUE TITLE
FSQ GPU hang out when devices > 1 `get_maybe_sync_seed`

### ISSUE THREAD

**Issue by ChuxiJ** [lucidrains/vector-quantize-pytorch#198]
https://github.com/lucidrains/vector-quantize-pytorch/blob/4380fe8b7b4435c28544955639563a00db47e8e6/vector_quantize_pytorch/residual_fsq.py#L39

It will hang forever when I use pytorch_lighting train fsq vqvae with gpu devices>1.

After commenting out the following code, it works!

What the function of this part is. Does it matter for performance?

```python
def get_maybe_sync_seed(device, max_size = 10_000):
    rand_int = torch.randint(0, max_size, (), device = device)

    # if is_distributed():
    #     print("debnug before dist.all_reduce")
    #     dist.all_reduce(rand_int)

    return rand_int.item()
```

**Comment by lucidrains** [comment]
yea it does matter

did you fix it?

**Comment by HQYSS** [comment]
If you use all_reduce in inference mode or no_grad, this error will happen.

The all_reduce here is to set the random seed of each device to be the same number.
It can affect the performance of dropout.

But if you use it in inference mode, you don't need to care about dropout, hence you can just comment out it.
